tools/buildsys/src/main.rs

```diff
@@ -270,6 +274,86 @@ fn build_variant() -> Result<()> {
     Ok(())
 }
 
+// Get all repo paths that make up the source directories for a given variant.
+// Expects BUIDLSYS_ROOT_DIR and BUILDSYS_VARIANT environment variables to be
+// set to know the path to the root of the repo and the variant to inspect.
+fn variant_paths() -> Result<()> {
+    let root_dir: PathBuf = getenv("BUILDSYS_ROOT_DIR")?.into();
+    let root_dir = root_dir.canonicalize().unwrap_or(root_dir);
+
+    let variant = getenv("BUILDSYS_VARIANT")?;
+    let manifest_dir = root_dir.join("variants").join(variant);
+
+    let manifest =
+        ManifestInfo::new(manifest_dir.join(MANIFEST_FILE)).context(error::ManifestParseSnafu)?;
+
+    let mut results: HashSet<PathBuf> = HashSet::new();
+
+    // manifest.included_packages gives the RPM package names, which doesn't always match with the local
+    // package directory name. We need to use the [*dependencies] sections to find the real packages.
+    results.insert(manifest_dir.clone());
```

Comment:
Might be better to add a `ManifestInfo::preferred_name()` function that either returns `package.metadata.build-package.package-name` (if defined) or else `package.name`. And then update `build_package` to use that also.